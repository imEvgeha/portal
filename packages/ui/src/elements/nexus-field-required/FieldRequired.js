import React from 'react';
import PropTypes from 'prop-types';

const FieldRequired = ({required}) => {
    return required ? <span className="p-field-required">*</span> : null;
};
FieldRequired.propTypes = {
    required: PropTypes.bool,
};

FieldRequired.defaultProps = {
    required: false,
};

export default FieldRequired;
