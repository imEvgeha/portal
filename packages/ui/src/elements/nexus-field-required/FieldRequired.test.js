import React from 'react';
import {shallow} from 'enzyme';
import FieldRequired from './FieldRequired';

describe('FieldRequired', () => {
    let wrapper = null;

    beforeEach(() => {
        wrapper = shallow(<FieldRequired required={true} />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should be required', () => {
        expect(wrapper.find('.p-field-required')).toHaveLength(1);
    });
});
