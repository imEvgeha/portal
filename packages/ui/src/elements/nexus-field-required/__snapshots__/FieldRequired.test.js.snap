// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldRequired should match snapshot 1`] = `
<span
  className="p-field-required"
>
  *
</span>
`;
